src/AstroImages.jl

```diff
@@ -85,9 +85,21 @@ for n in (8, 16, 32, 64)
     end
 end
 
+mutable struct properties{T <: AbstractArray}
+    rgb_image::T
```

Comment:
Is this always `Array{T,3}`?  In this case, the type of this field can be `Array{T,3}` and the parameter of the type just `T`.

Reply:
It's type is usually like this `MappedArrays.MultiMappedArray{RGB{Float64},2,Tuple{Array{Float64,2},Array{Float64,2},Array{Float64,2}},Type{RGB{Float64}},typeof(ImageCore.extractchannels)}`, which is a subtype of `AbstractArray`. So it won't be right to write `Array{T,3}` there.

Reply:
Then use something like `MappedArrays.MultiMappedArray{RGB{T},2,Tuple{Array{T,2},Array{T,2},Array{T,2}},Type{RGB{T}},S}`.  Just make it fixed what is always fixed, and make it a parameter what can change